src/ipyradiant/visualization/interactive_exploration.py

```diff
@@ -0,0 +1,309 @@
+        # Here we have to set the children again so that the changes propogate to the front end
+        # automatically. Ideally this would be done with traits but did not seem to work. LINK TO GITHUB ISSUE:
+        # https://github.com/jupyrdf/ipyradiant/issues/79
+        self.children = (
+            self.cyto_graph,
+            W.HBox(
+                children=[
+                    self.expand_button,
+                    self.undo_button,
+                    self.remove_temp_nodes_button,
+                ]
+            ),
+        )
+
+    @trt.validate("children")
+    def validate_children(self, proposal):
+        children = proposal.value
+        if not children:
+            children = (
+                self.cyto_graph,
+                W.HBox(
+                    children=[
+                        self.expand_button,
+                        self.undo_button,
+                        self.remove_temp_nodes_button,
+                    ]
+                ),
+            )
+        return children
+
+    def get_node(self, node: Node) -> Node:
+        """
+        This function is used to find a node given the id of a node copy.
+        """
+
+        for node_obj in self.cyto_graph.graph.nodes:
+            if node_obj.data["id"] == node["data"]["id"]:
+                return node_obj
+        # TODO: Make this function return None and log a warning if not node not found.
+        raise ValueError("Node not found in cytoscape.graph.nodes.")
+
+    def log_node_clicks(self, node: Node):
+        """
+        This function works with registering a click on a node. This will mark the node as selected and change the color of the
+        selected node.
+        """
+
+        node_object = self.get_node(node)
+
+        if self.selected_node == node_object:
+            node_object.classes = "clicked"
+            # NOTE: Class changes won't propogate to the front end for added nodes until
+            # the graph is updated.
+            # To fix this we create a random node and then quickly delete it so that the changes propogate.
+            self.update_cytoscape_frontend()
+
+        self.selected_node = node_object
+
+    def expand_button_clicked(self, button):
+        """
+        This function expands a node by loading in its predicates and subjects when
+        a node is selected and the expand button is clicked.
+        """
+        self.undo_button.disabled = False
+        if self.selected_node is None:
+            return None
+        new_data = GetOutgoingPredicateObjects.run_query(
+            graph=self.rdf_graph, s=self.selected_node.data["iri"]
+        )
+        objs = new_data["o"].tolist()
+        preds = new_data["p"].tolist()
+        labels = new_data["label"].tolist()
+        # add nodes
+        self.existing_node_ids = [
+            node.data["id"] for node in self.cyto_graph.graph.nodes
+        ]
+        self.new_nodes = {}
+        self.new_edges = {}
```

Comment:
```python

self.new_edges = {
    Edge(
        data={
            "source": self.selected_node.data["id"],
            "target": str(object_iri),
            "iri": URItoID(preds[object_idx]),
        },
        classes="temp",
    )
    for object_idx, object_iri in enumerate(objs)
}
```

`self.cyto_graph.graph.add_edges_from(self.new_edges.values())` (if it works, see note above)